### DISCUSSION TITLE
Apache reverse proxy with letsencrypt support (/.well-known/acme-challenge support)

### DISCUSSION THREAD

**Discussion by Eisteed** [photoprism/photoprism#1655] [Q&A] [answered]
Hello, I did setup photoprism with apache but the proxypass / rewrite rules make it impossible for letsencrypt to challenge the domain (and so get SSL certificate from letsencrypt)
After multiple test, here is an edited apache config that make it work for anyone interested :

```
<VirtualHost *:80>
        ServerName photoprism.domain
        #Https only
  	Redirect permanent / https://photoprism.domain/
</VirtualHost>

<VirtualHost *:443>
    ServerName photoprism.domain
    DocumentRoot /yourpath/to/public_html

    # Rest of your apache conf

    ProxyPreserveHost On
    ProxyPass /.well-known !
    RewriteCond %{HTTP:Upgrade} websocket [NC]
    RewriteCond %{HTTP:Connection} upgrade [NC]
    RewriteCond %{REQUEST_URI} !^/\.well\-known/acme\-challenge/
    RewriteRule ^/?(.*) "ws://127.0.0.1:2342/$1" [P,L]

    ProxyPass / http://127.0.0.1:2342/
    ProxyPassReverse / http://127.0.0.1:2342/
</VirtualHost>
```

**Comment by lastzero** [comment, marked as answer]
We recommend Traefik or Caddy for automatically updated Let's Encrypt HTTPS certificates:

- https://docs.photoprism.org/getting-started/proxies/traefik/
- https://docs.photoprism.org/getting-started/proxies/caddy-2/

Note you don't need to use port 80 for PhotoPrism. Modern browsers often try port 443 on their own. You can run other services on this port or use it for Let's Encrypt if Apache can't get a certificate otherwise. You won't need an open port either when using the DNS challenge for Let's Encrypt.

**Comment by TheNomad11** [comment]
Thanks a lot, it works perfectly. I  had already installed Apache on the VPS, so I was happy to find this apache reverse proxy configuration!
